### PR TITLE
[HIP][HIPIFY] Add some missing flags for cooperative launch and occup…

### PR DESCRIPTION
**Pull request by asalmanp**
…ancy APIs

### CODE REVIEW

File: bin/hipify-perl [hunk 1]
```diff
@@ -1801,6 +1803,9 @@ sub simpleSubstitutions {
     $ft{'define'} += s/\bcudaTextureType3D\b/hipTextureType3D/g;
     $ft{'define'} += s/\bcudaTextureTypeCubemap\b/hipTextureTypeCubemap/g;
     $ft{'define'} += s/\bcudaTextureTypeCubemapLayered\b/hipTextureTypeCubemapLayered/g;
+    $ft{'define'} += s/\bcudaOccupancyDefault\b/hipOccupancyDefault/g;
+    $ft{'define'} += s/\bcudaCooperativeLaunchMultiDeviceNoPreSync\b/hipCooperativeLaunchMultiDeviceNoPreSync/g;
+    $ft{'define'} += s/\bcudaCooperativeLaunchMultiDeviceNoPostSync\b/hipCooperativeLaunchMultiDeviceNoPostSync/g;
```
Comment (emankov) [inline bin/hipify-perl]: It looks like hipify-perl changes were obtained by hand. Please, generate hipify-perl script by `hipify-clang -perl` and update the diff. hipify-clang should be built against your changes beforehand.

File: hipify-clang/src/CUDA2HIP_Runtime_API_types.cpp [hunk 2]
```diff
@@ -787,7 +787,7 @@ const std::map<llvm::StringRef, hipCounter> CUDA_RUNTIME_TYPE_NAME_MAP {
   {"cudaErrorInvalidPc",                                               {"hipErrorInvalidPc",                                        "", CONV_NUMERIC_LITERAL, API_RUNTIME, HIP_UNSUPPORTED}}, // 718
   // CUDA_ERROR_LAUNCH_FAILED
   {"cudaErrorLaunchFailure",                                           {"hipErrorLaunchFailure",                                    "", CONV_NUMERIC_LITERAL, API_RUNTIME}}, // 719
-  // no analogue
+  // CUDA_ERROR_COOPERATIVE_LAUNCH_TOO_LARGE
   {"cudaErrorCooperativeLaunchTooLarge",                               {"hipErrorCooperativeLaunchTooLarge",                        "", CONV_NUMERIC_LITERAL, API_RUNTIME, HIP_UNSUPPORTED}}, // 720
```
Comment (emankov) [inline hipify-clang/src/CUDA2HIP_Runtime_API_types.cpp]: Please remove `HIP_UNSUPPORTED`

File: hipify-clang/src/CUDA2HIP_Driver_API_types.cpp [hunk 3]
```diff
@@ -1284,6 +1284,8 @@ const std::map<llvm::StringRef, hipCounter> CUDA_DRIVER_TYPE_NAME_MAP{
   {"CUDA_ERROR_INVALID_PC",                                            {"hipErrorInvalidPc",                                        "", CONV_NUMERIC_LITERAL, API_DRIVER, HIP_UNSUPPORTED}}, // 718
   // cudaErrorLaunchFailure
   {"CUDA_ERROR_LAUNCH_FAILED",                                         {"hipErrorLaunchFailure",                                    "", CONV_NUMERIC_LITERAL, API_DRIVER}}, // 719
+  // cudaErrorCooperativeLaunchTooLarge
+  {"CUDA_ERROR_COOPERATIVE_LAUNCH_TOO_LARGE",                          {"hipErrorCooperativeLaunchTooLarge",                        "", CONV_NUMERIC_LITERAL, API_DRIVER, HIP_UNSUPPORTED}}, // 720
```
Comment (emankov) [inline hipify-clang/src/CUDA2HIP_Driver_API_types.cpp]: Please remove `HIP_UNSUPPORTED`

File: docs/markdown/CUDA_Driver_API_functions_supported_by_HIP.md [hunk 4]
```diff
@@ -1063,8 +1064,8 @@
 | `cuFuncSetSharedMemConfig`                                |                               |
 | `cuLaunchKernel`                                          | `hipModuleLaunchKernel`       |
 | `cuLaunchHostFunc`                                        |                               | 10.0             |
-| `cuLaunchCooperativeKernel`                               |                               | 9.0              |
-| `cuLaunchCooperativeKernelMultiDevice`                    |                               | 9.0              |
+| `cuLaunchCooperativeKernel`                                | `hipLaunchCooperativeKernel` | 9.0              |
+| `cuLaunchCooperativeKernelMultiDevice`                    | `hipLaunchCooperativeKernelMultiDevice` | 9.0 |
```
Comment (emankov) [inline docs/markdown/CUDA_Driver_API_functions_supported_by_HIP.md]: `cuLaunchCooperativeKernel` is not equal to `cudaLaunchCooperativeKernel `due to different signatures; the same is for `cuLaunchCooperativeKernelMultiDevice`.

File: include/hip/nvcc_detail/hip_runtime_api.h [hunk 5]
```diff
@@ -140,6 +140,12 @@ typedef enum cudaChannelFormatKind hipChannelFormatKind;
 #define hipLimitMallocHeapSize cudaLimitMallocHeapSize
 #define hipIpcMemLazyEnablePeerAccess cudaIpcMemLazyEnablePeerAccess
 
+#define hipOccupancyDefault cudaOccupancyDefault
+
+#define hipCooperativeLaunchMultiDeviceNoPreSync cudaCooperativeLaunchMultiDeviceNoPreSync
+#define hipCooperativeLaunchMultiDeviceNoPostSync cudaCooperativeLaunchMultiDeviceNoPostSync
+
+
```
Comment (emankov) [inline include/hip/nvcc_detail/hip_runtime_api.h]: Excessive line.

File: include/hip/hcc_detail/hip_runtime_api.h [hunk 6]
```diff
@@ -2917,7 +2922,7 @@ hipError_t hipLaunchCooperativeKernel(const void* f, dim3 gridDim, dim3 blockDim
  * @param [in] numDevices               Size of the launchParamsList array.
  * @param [in] flags                    Flags to control launch behavior.
  *
- * @returns hipSuccess, hipInvalidDevice, hipErrorNotInitialized, hipErrorInvalidValue
+ * @returns hipSuccess, hipInvalidDevice, hipErrorNotInitialized, hipErrorInvalidValue, hipErrorCooperativeLaunchTooLarge
```
Comment (emankov) [inline include/hip/hcc_detail/hip_runtime_api.h]: Where are the corresponding changes in `hipLaunchCooperativeKernelMultiDevice` and `hipLaunchCooperativeKernel` functions implementation, which handle erroneous situations with emitting `hipErrorCooperativeLaunchTooLarge` error code?

File: hipify-clang/src/CUDA2HIP_Runtime_API_types.cpp (hunk 2 above)
Comment (asalmanp) [reply hipify-clang/src/CUDA2HIP_Runtime_API_types.cpp]: removed HIP_UNSUPPORTED

File: docs/markdown/CUDA_Driver_API_functions_supported_by_HIP.md (hunk 4 above)
Comment (asalmanp) [reply docs/markdown/CUDA_Driver_API_functions_supported_by_HIP.md]: modified the doc.

File: include/hip/hcc_detail/hip_runtime_api.h (hunk 6 above)
Comment (asalmanp) [reply include/hip/hcc_detail/hip_runtime_api.h]: The corresponding changes in cooperative launch APIs are coming in a separate PR next. The purpose of this PR is to just add the missing flags in the headers.

File: hipify-clang/src/CUDA2HIP_Driver_API_types.cpp (hunk 3 above)
Comment (asalmanp) [reply hipify-clang/src/CUDA2HIP_Driver_API_types.cpp]: removed HIP_UNSUPPORTED

File: bin/hipify-perl [hunk 7]
```diff
@@ -1803,9 +1801,6 @@ sub simpleSubstitutions {
     $ft{'define'} += s/\bcudaTextureType3D\b/hipTextureType3D/g;
     $ft{'define'} += s/\bcudaTextureTypeCubemap\b/hipTextureTypeCubemap/g;
     $ft{'define'} += s/\bcudaTextureTypeCubemapLayered\b/hipTextureTypeCubemapLayered/g;
-    $ft{'define'} += s/\bcudaOccupancyDefault\b/hipOccupancyDefault/g;
-    $ft{'define'} += s/\bcudaCooperativeLaunchMultiDeviceNoPreSync\b/hipCooperativeLaunchMultiDeviceNoPreSync/g;
-    $ft{'define'} += s/\bcudaCooperativeLaunchMultiDeviceNoPostSync\b/hipCooperativeLaunchMultiDeviceNoPostSync/g;
```
Comment (emankov) [inline bin/hipify-perl]: `hipify-perl` should be always in sync with `hipify-clang`.

Comment (asalmanp) [reply bin/hipify-perl]: updated the hipify-perl

File: bin/hipify-perl (hunk 1 above)
Comment (asalmanp) [reply bin/hipify-perl]: generated the new hipify-perl

**Comment by emankov** [comment]
@asalmanp, could you please rebase your change?

**Comment by asalmanp** [comment]
@emankov @mangupta I rebased my change, the CI is green now. can you please take a look one more time and help to merge it to master branch?

**Pull request merged**